mock/mock.go

```diff
@@ -798,6 +832,29 @@ func (args Arguments) Diff(objects []interface{}) (string, int) {
 				differences++
 				output = fmt.Sprintf("%s\t%d: FAIL:  type %s != type %s - %s\n", output, i, reflect.TypeOf(t).Name(), reflect.TypeOf(actual).Name(), actualFmt)
 			}
+		} else if reflect.TypeOf(expected) == reflect.TypeOf((*FunctionalOptionsArgument)(nil)) {
+			t := expected.(*FunctionalOptionsArgument).value
+
+			var name string
+			tValue := reflect.ValueOf(t)
+			if tValue.Len() > 0 {
+				name = "[]" + reflect.TypeOf(tValue.Index(0).Interface()).String()
+			}
+
+			tName := reflect.TypeOf(t).Name()
+			if name != reflect.TypeOf(actual).String() && tValue.Len() != 0 {
+				differences++
+				output = fmt.Sprintf("%s\t%d: FAIL:  type %s != type %s - %s\n", output, i, tName, reflect.TypeOf(actual).Name(), actualFmt)
```

Comment:
I believe I'm following the test setup nearly the same (options are from another package) and I noticed 2 things.
1. `tName` is empty for me but `reflect.TypeOf(tValue.Index(0).Interface()).String()` has the same type as `reflect.TypeOf(actual).String()`
2. `name` has the prefix `[]` so I'm hitting this branch

Curious the choice for `tName` and `reflect.TypeOf(actual).Name()` in the log here instead of `name` and `reflect.TypeOf(actual).String()` that is used in the conditional.

It was hard for me to debug as the logs showed empty string is not the same type as my option type. When debugging I found that `name` was really close except for the prefixed `[]`

I might follow up with a PR to fix the logged out line here if it was just an accident.